### ISSUE TITLE
[Clang 8 + -O2] Crash with uncommon aspect ratio or -enc-mode > 2

### ISSUE THREAD

**Issue by jbeich** [OpenVisualCloud/SVT-AV1#151]
8d75aadc3a3c snapshot, built with Clang 8.0.0 (rc3) on FreeBSD. GCC 8.2, Clang 7.0/6.0 are not affected. Fixed by #124.

```
$ curl -O https://media.xiph.org/video/derf/y4m/bus_cif.y4m
$ SvtAv1EncApp -i bus_cif.y4m -b bus_cif.ivf
-------------------------------------------
SVT-AV1 Encoder
SVT [version]:  SVT-AV1 Encoder Lib v0.4.0
SVT [build]  :  GCC 4.2.1        64 bit
LIB Build date: Mar  6 2019 12:24:19
-------------------------------------------
Number of logical cores available: 8
Number of PPCS 41
-------------------------------------------
SVT [config]: Main Profile      Tier (auto)     Level (auto)
SVT [config]: EncoderMode                                                       : 7
SVT [config]: EncoderBitDepth / CompressedTenBitFormat                          : 8 / 0
SVT [config]: SourceWidth / SourceHeight                                        : 352 / 288
SVT [config]: Fps_Numerator / Fps_Denominator / Gop Size / IntraRefreshType     : 30 / 1 / 32 / 1
SVT [config]: HierarchicalLevels / BaseLayerSwitchMode / PredStructure          : 4 / 0 / 2
SVT [config]: BRC Mode / QP  / LookaheadDistance / SceneChange                  : CQP / 50 / 33 / 0
-------------------------------------------
Encoding          Assertion failed: (mi_size_wide_log2[bsize] == mi_size_high_log2[bsize]), function EncodePartitionAv1, file Source/Lib/Codec/EbEntropyCoding.c, line 994.

Thread 63 received signal SIGABRT, Aborted.
[Switching to LWP 102093 of process 66613]
thr_kill () at thr_kill.S:3
3       thr_kill.S: No such file or directory.
(gdb) backtrace full
#0  thr_kill () at thr_kill.S:3
No locals.
#1  0x00000008019e6fe4 in __raise (s=6) at /usr/src/lib/libc/gen/raise.c:52
        id = 102093
#2  0x0000000801948969 in abort () at /usr/src/lib/libc/stdlib/abort.c:67
        act = <optimized out>
#3  0x00000008019d1881 in __assert (func=<optimized out>, file=<optimized out>, line=<optimized out>,
    failedexpr=<optimized out>) at /usr/src/lib/libc/gen/assert.c:51
No locals.
#4  0x000000080145a3cc in EncodePartitionAv1 (sequence_control_set_ptr=0x85a7a0600, frameContext=0x867e78600,
    ecWriter=0x85e0ab070, bsize=<optimized out>, p=<optimized out>, cu_origin_x=<optimized out>,
    cu_origin_y=<optimized out>, partition_context_neighbor_array=0x867c24bd0)
    at Source/Lib/Codec/EbEntropyCoding.c:994
        contextIndex = <optimized out>
        bsl = <optimized out>
        partitionContextLeftNeighborIndex = 0
        hbs = <optimized out>
        partitionContextTopNeighborIndex = <optimized out>
        left = <optimized out>
        above = <optimized out>
        is_partition_point = <optimized out>
        hasRows = <optimized out>
        hasCols = <optimized out>
        above_ctx = <optimized out>
        left_ctx = <optimized out>
        cdf = <optimized out>
        cdf = <optimized out>
#5  write_sb (context_ptr=<optimized out>, tbPtr=0x8536dce00, picture_control_set_ptr=<optimized out>,
    entropy_coder_ptr=0x85e0ab060, coeffPtr=0x8536dce60)
    at Source/Lib/Codec/EbEntropyCoding.c:5535
        cm = 0x85e0b9000
        blkOriginX = <optimized out>
        blkOriginY = <optimized out>
        hbs = <optimized out>
        mi_row = 0
        mi_col = 80
        quarter_step = <optimized out>
        codeCuCond = <optimized out>
        frameContext = 0x867e78600
        ecWriter = 0x85e0ab070
        sequence_control_set_ptr = 0x85a7a0600
        partition_context_neighbor_array = 0x867c24bd0
        final_cu_index = 0
        cu_index = 26
        checkCuOutOfBound = 0 '\000'
        cu_ptr = 0x868cac480
        blk_geom = <optimized out>
        cu_origin_x = <optimized out>
        cu_origin_y = <optimized out>
        bsize = <optimized out>
        sb_geom = <optimized out>
#6  0x000000080145f968 in EntropyCodingLcu (context_ptr=<optimized out>, sb_ptr=<optimized out>,
    picture_control_set_ptr=0x867bad000, sequence_control_set_ptr=<optimized out>, sb_origin_x=<optimized out>,
    sb_origin_y=<optimized out>, terminateSliceFlag=<optimized out>, pictureOriginX=<optimized out>,
    pictureOriginY=<optimized out>)
    at Source/Lib/Codec/EbEntropyCodingProcess.c:393
        coeffPicturePtr = 0x8536dce60
        writtenBitsBeforeQuantizedCoeff = 0
        writtenBitsAfterQuantizedCoeff = <optimized out>
#7  EntropyCodingKernel (input_ptr=<optimized out>)
    at Source/Lib/Codec/EbEntropyCodingProcess.c:597
        rowTotalBits = 0
        context_ptr = <optimized out>
        picture_width_in_sb = 6
        picture_control_set_ptr = 0x867bad000
        sequence_control_set_ptr = 0x85a7a0600
        encDecResultsWrapperPtr = 0x87fa6ce80
        encDecResultsPtr = <optimized out>
        lastLcuFlag = 0 '\000'
        initialProcessCall = 1 '\001'
        yLcuIndex = <optimized out>
        xLcuIndex = 5
        sb_origin_y = 0
        sb_origin_x = <optimized out>
        sb_ptr = 0x8536dce00
        entropyCodingResultsPtr = <optimized out>
        entropyCodingResultsWrapperPtr = <optimized out>
        sb_index = <optimized out>
        sb_sz = <optimized out>
        lcuSizeLog2 = <optimized out>
#8  0x0000000801088776 in thread_start (curthread=0x89e1dd600) at /usr/src/lib/libthr/thread/thr_create.c:291
        set = {__bits = {0, 0, 0, 0}}
#9  0x0000000000000000 in ?? ()
No symbol table info available.
Backtrace stopped: Cannot access memory at address 0x7fffd81bf000
```

**Comment by hassount** [comment]
is this still valid ? Travis seems to be passing with clang

**Comment by luigi311** [comment]
@hassount 
I believe this is ready to be closed since he said 
>Fixed by #124.

**Comment by jbeich** [comment]
I can't reproduce. When the issue was filed the fix was **known** but **merge time** wasn't.
